internal/support/scheduler/utils.go

```diff
@@ -44,7 +49,19 @@ func msToTime(ms string) (time.Time, error) {
 // Hour = 60 * Minute
 func parseFrequency(durationStr string) (time.Duration, error) {
 
-	// check Frequency
+	// Legacy ISO8601 format P1
+	// ^P(\d+Y)?(\d+M)?(\d+D)?(T(\d+H)?(\d+M)?(\d+S)?)?$`
+	matched, _ := regexp.MatchString(frequencyPattern, durationStr)
```

Comment:
- Can you please point out which new field(s), added to the interval action, regarding `schedulecontext.go` were updated leading up to the issue that resulted in this PR?

Reply:
@xmlviking Following our chat, we're good here.

Reply:
@akramtexas @xmlviking Can ya'll document your conversation here as part of the issue for everyone else's benefit and posterity's sake?

Reply:
@michaelestrin Sure, will do 👍

Reply:
There are no additional fields added to the IntervalAction or Interval which affected this particular issue.  I'm sorry if added to any confusion around that.  The issue as I saw it was related to the inclusion of the parseFrequency() function in the scheduler utils.go file.  It did validate both the "legacy frequency" format as well as the new GoLang style, however it did not actually convert the "legacy" time string format into a proper time.Duration.  When creating new SchedulerContext instances (Interval + 1 to N IntervalActions) the frequency field was not going to work in the legacy format.   This fix ensures that if the frequency field is in either formats "legacy" or new "golang time duration style" it will work.

Reply:
To add (a few supporting details) to that, the existing `func parseFrequency(durationStr string) (time.Duration, error) { ... }` has been augmented with the following call:
```
timeDuration := parseLegacyFrequency(durationStr)
```
That call (to `func parseLegacyFrequency(durationStr string) time.Duration { ... }`) handles the `legacy` approach of parsing frequency (aka `interval`).